src/Dialog/Dialog.stories.tsx

```diff
@@ -0,0 +1,202 @@
+import React, { PropsWithChildren, useEffect, useRef, useState } from 'react'
+import { Story } from '@storybook/react'
+import styled from 'styled-components'
+
+import { Grid, Section } from '../'
+import { default as DialogComponent } from './index'
+
+export default {
+  title: 'Components/Dialog',
+  component: DialogComponent,
+  argTypes: {
+    includeClose: {
+      control: {
+        type: 'boolean',
+      },
+    },
+    borderRadius: {
+      control: {
+        type: 'text',
+      },
+    },
+    padding: {
+      control: {
+        type: 'text',
+      },
+    },
+    margin: {
+      control: {
+        type: 'text',
+      },
+    },
+    border: {
+      control: {
+        type: 'text',
+      },
+    },
+    boxShadow: {
+      control: {
+        type: 'text',
+      },
+    },
+    width: {
+      control: {
+        type: 'text',
+      },
+    },
+    height: {
+      control: {
+        type: 'text',
+      },
+    },
+  },
+}
+
+const DialogStoryTemplate: Story = (args) => {
+  return <DefaultStoryHost {...args} useModal={false} />
+}
+export const Dialog = DialogStoryTemplate.bind({})
+Dialog.args = {
+  includeClose: true,
+  modalBackdropColor: 'rgba(255, 0, 0, 0.05)',
+  borderRadius: '20px',
+  padding: '0',
+  margin: '0 auto',
+  border: '1px solid black',
+  boxShadow: '5px 5px 5px #90909090',
+  width: 'fit-content',
+  height: 'auto',
+}
+
+const ModalStoryTemplate: Story = (args) => {
+  return <DefaultStoryHost {...args} useModal={true} />
+}
+export const Modal = ModalStoryTemplate.bind({})
+Modal.args = {
+  includeClose: true,
+  modalBackdropColor: 'rgba(255, 0, 0, 0.05)',
+  borderRadius: '20px',
+  padding: '0',
+  margin: '0 auto',
+  border: '0',
+  boxShadow: '5px 5px 5px #90909090',
+  width: 'fit-content',
+  height: 'fit-content',
+}
+
+const DefaultStoryHost: React.FC<any> = ({ useModal, ...args }) => {
+  const dialogRef = useRef<HTMLDialogElement>(null)
+  const [isOpen, setIsOpen] = useState(false)
+
+  useEffect(() => {
+    const listener = () => {
+      setIsOpen(false)
+    }
+    const dialog = dialogRef.current
+    dialog?.addEventListener('close', listener)
+
+    return () => dialog?.removeEventListener('close', listener)
+  })
+
+  return (
+    <div>
+      <button
+        onClick={() => {
+          setIsOpen(!isOpen)
+          if (!isOpen) {
+            if (useModal) dialogRef.current?.showModal()
+            else dialogRef.current?.show()
+          } else dialogRef.current?.close()
+        }}
+      >
+        {!isOpen ? 'Open' : 'Close'}
+      </button>
+      <DialogComponent ref={dialogRef} {...args}>
+        <MockContent />
+      </DialogComponent>
+    </div>
+  )
+}
+
+const MockContent = styled.div`
+  margin: 2ch;
+  background: cyan;
+  width: 300px;
+  height: 300px;
+`
+
+export const HII: Story = () => {
```

Comment:
it would be nice if we would not need to use grid when using this component, e.g. what we wrap dialog comp into. Maybe even default to be absolute but relative to the parent if it makes sense... not sure is just something came into my mind

Reply:
I think there's a HII specific component in this. But dialog shouldn't make any assumptions about it's content IMHO